rustls/src/msgs/handshake.rs

```diff
@@ -1029,9 +1037,11 @@ impl ClientHelloPayload {
     }
 
     pub fn set_psk_binder(&mut self, binder: impl Into<Vec<u8>>) {
-        let last_extension = self.extensions.last_mut();
-        if let Some(ClientExtension::PresharedKey(ref mut offer)) = last_extension {
-            offer.binders[0] = PresharedKeyBinder::new(binder.into());
+        if let Some(ref mut extensions) = &mut self.extensions {
```

Comment:
I'd probably write these as `match` expressions with an early return clause in the `None arm.